<launch>

  <arg name="robot_model_master"                default="rx200"/>
  <arg name="robot_model_puppet"                default="rx200"/>

  <arg name="base_link_master"                  default="base_link"/>
  <arg name="base_link_puppet"                  default="base_link"/>

  <arg name="master_modes"                      default="$(find robot_teleop)/config/master_modes.yaml"/>
  <arg name="puppet_modes"                      default="$(find robot_teleop)/config/puppet_modes.yaml"/>

  <arg name="launch_driver"                     default="true"/>
  <arg name="use_sim"                           default="false"/>

  <arg name="robot_name_master"                 value="master"/>
  <arg name="robot_name_puppet"                 value="puppet"/>


  <include if="$(arg launch_driver)" file="$(find interbotix_xsarm_control)/launch/xsarm_control.launch">
    <arg name="robot_model"                       value="$(arg robot_model_master)"/>
    <arg name="robot_name"                        value="$(arg robot_name_master)"/>
    <arg name="base_link_frame"                   value="$(arg base_link_master)"/>
    <arg name="use_world_frame"                   value="false"/>
    <arg name="use_rviz"                          value="false"/>
    <arg name="mode_configs"                      value="$(arg master_modes)"/>
    <arg name="use_sim"                           value="$(arg use_sim)"/>
  </include>

  <include if="$(arg launch_driver)" file="$(find interbotix_xsarm_control)/launch/xsarm_control.launch">
    <arg name="robot_model"                       value="$(arg robot_model_puppet)"/>
    <arg name="robot_name"                        value="$(arg robot_name_puppet)"/>
    <arg name="base_link_frame"                   value="$(arg base_link_puppet)"/>
    <arg name="use_world_frame"                   value="false"/>
    <arg name="use_rviz"                          value="false"/>
    <arg name="mode_configs"                      value="$(arg puppet_modes)"/>
    <arg name="use_sim"                           value="$(arg use_sim)"/>
  </include>


  <node
    name="master_transform_broadcaster"
    pkg="tf2_ros"
    type="static_transform_publisher"
    args="0 -0.25 0 0 0 0 /world /$(arg robot_name_master)/base_link"/>

  <node
    name="puppet_transform_broadcaster"
    pkg="tf2_ros"
    type="static_transform_publisher"
    args="0 0.25 0 0 0 0 /world /$(arg robot_name_puppet)/base_link"/>

  <include if="$(arg launch_driver)" file="$(find realsense2_camera)/launch/rs_camera.launch">
    <arg name="camera" value="cam_high" />
    <arg name="enable_color" value="true" />
    <arg name="color_width" value="640" />
    <arg name="color_height" value="480" />
    <arg name="color_fps" value="60"/>
  </include>

</launch>
